rdkit/Chem/Draw/UnitTestDraw.py

```diff
@@ -213,6 +255,122 @@ def testGridSVG(self):
+
+    ## Test that exceptions are thrown appropriately
+
+    # Test that supplying a non-nested list raises a ValueError
+    # Set up non-nested lists = first sublist of nested lists
+    mols_not_nested = self.mols_matrix[0]
+    legends_not_nested = self.legends_matrix[0]
+    highlightAtomLists_not_nested = self.highlightAtomLists_matrix[0]
+    highlightBondLists_not_nested = self.highlightBondLists_matrix[0]
+
+    with self.assertRaises(ValueError):
+      Draw._MolsNestedToLinear(mols_not_nested)
+
+    with self.assertRaises(ValueError):
+      Draw._MolsNestedToLinear(mols_matrix=self.mols_matrix, legends_matrix=legends_not_nested)
+
+    with self.assertRaises(ValueError):
+      Draw._MolsNestedToLinear(mols_matrix=self.mols_matrix, highlightAtomLists_matrix=highlightAtomLists_not_nested)
+
+    with self.assertRaises(ValueError):
+      Draw._MolsNestedToLinear(mols_matrix=self.mols_matrix, highlightBondLists_matrix=highlightBondLists_not_nested)
+
+    # Test that raises ValueError if other matrices aren't same size (# rows) as mols_matrix
+    with self.assertRaises(ValueError):
+      Draw._MolsNestedToLinear(mols_matrix=self.mols_matrix, legends_matrix=self.legends_matrix[0:1])
+
+    with self.assertRaises(ValueError):
+      Draw._MolsNestedToLinear(mols_matrix=self.mols_matrix, highlightAtomLists_matrix=self.highlightAtomLists_matrix[0:1])
+
+    with self.assertRaises(ValueError):
+      Draw._MolsNestedToLinear(mols_matrix=self.mols_matrix, highlightBondLists_matrix=self.highlightBondLists_matrix[0:1])
+
+    # Test that raises ValueError if other matrices' rows aren't same length as mols_matrix's corresponding row
+    # Remove last element from first row of each other matrix
+    legends_matrix_short_row0 = [self.legends_matrix[0][0:-1]] + [self.legends_matrix[1:]]
+    highlightAtomLists_matrix_short_row0 = [self.highlightAtomLists_matrix[0][0:-1]] + [self.highlightAtomLists_matrix[1:]]
+    highlightBondLists_matrix_short_row0 = [self.highlightBondLists_matrix[0][0:-1]] + [self.highlightBondLists_matrix[1:]]
+
+    with self.assertRaises(ValueError):
+      Draw._MolsNestedToLinear(mols_matrix=self.mols_matrix, legends_matrix=legends_matrix_short_row0)
+
+    with self.assertRaises(ValueError):
+      Draw._MolsNestedToLinear(mols_matrix=self.mols_matrix, highlightAtomLists_matrix=highlightAtomLists_matrix_short_row0)
+
+    with self.assertRaises(ValueError):
+      Draw._MolsNestedToLinear(mols_matrix=self.mols_matrix, highlightBondLists_matrix=highlightBondLists_matrix_short_row0)
+
+
+  def testMolsMatrixToGridImage(self):
+    # Tests are that running Draw.MolsMatrixToGridImage doesn't give an error
+    # Parametrize tests: In addition to supplying mols_matrix, supply 0-3 other matrices
+    subImgSize = (200, 200)
+
+    kwargs_value = Draw.rdMolDraw2D.MolDrawOptions()
+    kwargs_value.addAtomIndices = True
+
+    for (legends_matrix, highlightAtomLists_matrix, highlightBondLists_matrix) in self.param_sets:
```

Comment:
Just a comment: these loops aren't really testing anything other than "calling the function in this way doesn't produce an error". Testing drawing code is really hard, and this is a lot better than nothing.
I don't think you need to change anything, but it might be worth a comment at the top of the loop about this